pkg/controller/jobs/raycluster/raycluster_webhook.go

```diff
@@ -0,0 +1,129 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+)
+
+type RayClusterWebhook struct {
+	manageJobsWithoutQueueName bool
+}
+
+// SetupRayClusterWebhook configures the webhook for rayclusterapi RayCluster.
+func SetupRayClusterWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
+	options := jobframework.DefaultOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	wh := &RayClusterWebhook{
+		manageJobsWithoutQueueName: options.ManageJobsWithoutQueueName,
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&rayclusterapi.RayCluster{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-ray-io-v1-raycluster,mutating=true,failurePolicy=fail,sideEffects=None,groups=ray.io,resources=rayclusters,verbs=create,versions=v1,name=mraycluster.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &RayClusterWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *RayClusterWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*rayclusterapi.RayCluster)
+	log := ctrl.LoggerFrom(ctx).WithName("raycluster-webhook")
+	log.V(10).Info("Applying defaults", "job", klog.KObj(job))
+	jobframework.ApplyDefaultForSuspend((*RayCluster)(job), w.manageJobsWithoutQueueName)
+	return nil
+}
+
+// +kubebuilder:webhook:path=/validate-ray-io-v1-raycluster,mutating=false,failurePolicy=fail,sideEffects=None,groups=ray.io,resources=rayclusters,verbs=create;update,versions=v1,name=vraycluster.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &RayClusterWebhook{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (w *RayClusterWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	job := obj.(*rayclusterapi.RayCluster)
+	log := ctrl.LoggerFrom(ctx).WithName("raycluster-webhook")
+	log.V(10).Info("Validating create", "job", klog.KObj(job))
+	return nil, w.validateCreate(job).ToAggregate()
+}
+
+func (w *RayClusterWebhook) validateCreate(job *rayclusterapi.RayCluster) field.ErrorList {
+	var allErrors field.ErrorList
+	kueueJob := (*RayCluster)(job)
+
+	if w.manageJobsWithoutQueueName || jobframework.QueueName(kueueJob) != "" {
+		spec := &job.Spec
+		specPath := field.NewPath("spec")
+
+		// Should not want existing cluster. Keuue (workload) should be able to control the admission of the actual work, not only the trigger.
+		// ADD check
+
+		// Should not use auto scaler. Once the resources are reserved by queue the cluster should do it's best to use them.
+		if ptr.Deref(spec.EnableInTreeAutoscaling, false) {
```

Comment:
Add a TODO here to revisit this once https://github.com/kubernetes-sigs/kueue/issues/77 is implemented